includes/class-wc-payment-tokens.php

```diff
@@ -0,0 +1,249 @@
+	 */
+	public static function update( $token_id, $token ) {
+		$args = $token->__data_format();
+		unset( $args['token_id'] );
+		if ( ! self::validate( $args ) ) {
+			return; // @todo throw an error
+		}
+
+		// We need to separate our meta fields since they are stored in a separate table
+		$core_fields = self::get_token_core_fields();
+		$core_fields_to_create = array();
+		foreach ( $core_fields as $core_field ) {
+			if ( isset( $args[ $core_field ] ) ) {
+				$core_fields_to_update[ $core_field ] = $args[ $core_field ];
+				unset( $args[ $core_field ] );
+			}
+		}
+
+		global $wpdb;
+
+		// Update our core fields
+		$wpdb->update( $wpdb->prefix . 'woocommerce_payment_tokens', $core_fields_to_update, array( 'token_id' => $token_id ) );
+
+		// Grab a list of our existing meta, and update if we have a matching value in $args, otherise, create a new meta row
+		$existing_meta = $wpdb->get_col( $wpdb->prepare(
+			"SELECT meta_key FROM {$wpdb->prefix}woocommerce_payment_token_meta WHERE token_id = %d", $token_id
+		) );
+		foreach ( $args as $meta_key => $meta_value ) {
+			if ( in_array( $meta_key, $existing_meta ) ) {
+				$wpdb->update( $wpdb->prefix . 'woocommerce_payment_token_meta', array(
+					'meta_value' => $meta_value
+				), array( 'token_id'   => $token_id, 'meta_key'   => $meta_key ) );
+			} else {
+				$wpdb->insert( $wpdb->prefix . 'woocommerce_payment_token_meta', array(
+					'meta_key'   => $meta_key,
+					'meta_value' => $meta_value,
+					'token_id'   => $token_id
+				) );
+			}
+		}
+
+		do_action( 'woocommerce_payment_token_updated', $token_id );
+		return self::generate_token( $token_id, $core_fields_to_update );
+	}
+
+	/**
+	 * Remove a payment token from the database
+	 * @param int $token_id Token ID
+	 */
+	public static function delete( $token_id ) {
+		global $wpdb;
+		$wpdb->delete( $wpdb->prefix . 'woocommerce_payment_tokens', array( 'token_id' => $token_id ), array( '%d' ) );
+		$wpdb->delete( $wpdb->prefix . 'woocommerce_payment_token_meta', array( 'token_id' => $token_id ), array( '%d' ) );
+		do_action( 'woocommerce_payment_token_deleted', $token_id );
+	}
+
+	/**
+	 * Returns an array of payment token objects associated with the passed customer ID
+	 * @param int $customer_id Customer ID
+	 * @return array Array of token objects
+	 */
+	public static function get_customer_tokens( $customer_id ) {
+		if ( $customer_id < 1 ) {
+			return array();
+		}
+
+		global $wpdb;
+
+		$token_results = $wpdb->get_results( $wpdb->prepare(
+			"SELECT * FROM {$wpdb->prefix}woocommerce_payment_tokens WHERE customer_id = %d",
+			$customer_id
+		) );
+
+		if ( empty( $token_results ) ) {
+			return array();
+		}
+
+		$tokens = array();
+		foreach ( $token_results as $token_result ) {
+			$_token = self::generate_token( $token_result->token_id, (array) $token_result );
+			if ( ! empty( $_token ) ) {
+				$tokens[ $token_result->token_id ] = $_token;
+			}
+		}
+
+		return $tokens;
```

Comment:
I'd like to see a filter on this value so the local token data can be merged with the token data from a remote API call.